### PR TITLE
[EXE-1558] Setup build and deploy; add pushdown

### PR DESCRIPTION
**Pull request by dorisZ017**
* Upgrade sbt version and plugins; remove unused plugins; 
* Upgrade jdbc driver and hadoop version; 
* Upgrade to java 17; fix issues with test dependencies and compiling; 
* Setup deploying to jfrog
* Bringing in AIQ tracing and pushdowns in https://github.com/ActionIQ-OSS/spark-redshift/pull/9 and https://github.com/ActionIQ-OSS/spark-redshift/pull/10, also making them compatible with spark 3
* Exposing redshift statement in pushdown plan

Noticed some issue with partially invalid queries, documented here: https://actioniq.atlassian.net/browse/EXE-1614, will continue digging as part of bringing generic jdbc pushdown to flame.

Tested reading from redshift in https://github.com/ActionIQ/flame/pull/440/files

```
sc.setLogLevel("info")
// spark.conf.set("spark.aiq.sql.enable_jdbc_pushdown", true)
val redshiftOpts = Map(
  "aws_iam_role" -> "arn:aws:iam::938824509083:role/journeys-redshift-dev-redshift-role",
  "tempDir" -> s"s3a://aiq-1d-expire-dev/redshift/109",
  "driver" -> "com.amazon.redshift.jdbc42.Driver",
  "dbtable" -> "qahybridredshift.rd_dim_item",
  "queryGroup" -> "123456678908",
  "aiq_partner" -> "partnerName",
  "ApplicationName" -> "appName",
  "password" -> <>
  "user" -> <>,
  "url" -> "jdbc:redshift://test-hybridcompute.938824509083.us-east-1.redshift-serverless.amazonaws.com:5439/dev?"
)
val df = spark.read.format("io.github.spark_redshift_community.spark.redshift").options(redshiftOpts).load()
df.createOrReplaceTempView("t")
val qe = spark.sql("select color from t where lower(color) = 'snow'").queryExecution
qe.sparkPlan.asInstanceOf[io.github.spark_redshift_community.spark.redshift.pushdowns.RedshiftPushDownPlan].statement
res8: String = SELECT ( SUBQUERY_1.color ) AS SUBQUERY_2_COL_0 FROM ( SELECT * FROM ( SELECT * FROM qahybridredshift.rd_dim_item AS REDSHIFT_QUERY_ALIAS ) AS SUBQUERY_0 WHERE ( ( SUBQUERY_0.color IS NOT NULL ) AND ( LOWER ( SUBQUERY_0.color ) = ?(Some(snow)) ) ) ) AS SUBQUERY_1
```
Screenshot 2023-05-24 at 12 19 11 PM

```
sbt test
sbt publish
```

### CODE REVIEW

**Comment by MasterDDT** [comment]
Hmm how did Github let you request me for review if its still in draft mode 😄

**Comment by dorisZ017** [comment]
> Hmm how did Github let you request me for review if its still in draft mode 😄

No I was just adding the build and publish stuff but later figured I could just add pushdowns on top of it so converted it to draft... I split the pushdowns to another PR if you want to review the build part first

**Review by ShaoFuWu — COMMENTED** [review]
looks good 👍 one thing to follow up, can you test the pushdowns that we listed in our connector confluence and make sure those still work in spark 3, I guess can wait until you deploy spark and run from AIQ if thats easier.
Also, can you please fix the formatting in `RedshiftPushDownQuery.scala` please.

File: .jvmopts [hunk 1]
```diff
@@ -1,7 +1,4 @@
 -Dfile.encoding=UTF8
 -Xms1024M
 -Xmx1024M
--Xss6M
--XX:MaxPermSize=512m
```
Comment (MasterDDT) [inline .jvmopts:5]: These dont work with java 17?

Comment (dorisZ017) [reply .jvmopts:5]: They were removed in java 17 (or earlier): https://docs.oracle.com/en/java/javase/17/docs/specs/man/java.html#removed-java-options

File: build.sbt [hunk 2]
```diff
@@ -14,49 +14,47 @@
...
 
 // Define a custom test configuration so that unit test helper classes can be re-used under
 // the integration tests configuration; see http://stackoverflow.com/a/20635808.
 lazy val IntegrationTest = config("it") extend Test
 val testSparkVersion = sys.props.get("spark.testVersion").getOrElse(sparkVersion)
-val testHadoopVersion = sys.props.get("hadoop.testVersion").getOrElse("3.2.1")
+val testHadoopVersion = sys.props.get("hadoop.testVersion").getOrElse("3.3.2")
 // DON't UPGRADE AWS-SDK-JAVA if not compatible with hadoop version
-val testAWSJavaSDKVersion = sys.props.get("aws.testVersion").getOrElse("1.11.1033")
+val testAWSJavaSDKVersion = sys.props.get("aws.testVersion").getOrElse("1.11.1026")
```
Comment (MasterDDT) [inline build.sbt:32]: We use aws java 1.12.x, upgrade this one? Or is it just for tests and we dont care

File: src/main/scala/io/github/spark_redshift_community/spark/redshift/RedshiftJDBCPushdownWrapper.scala [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    }.mkString(",")
+
+    val finalQuery = if (aiqPropsString.nonEmpty) {
+      s"/* $aiqPropsString */\n$query"
+    } else {
+      query
+    }
+
+    if (jdbcProps.get("aiq_testing").exists(_.toBoolean)) {
+      RedshiftPushDownSqlStatement.capturedQueries.append(finalQuery)
```
Comment (MasterDDT) [inline src/main/scala/io/github/spark_redshift_community/spark/redshift/RedshiftJDBCPushdownWrapper.scala:90]: Prob dont need this anymore 🤷 , now that your exposed the statement in the plan. But I guess its fine to keep.

I dont see it used below in any tests?

**Review by MasterDDT — APPROVED** [review]

File: build.sbt (hunk 2 above)
Comment (dorisZ017) [reply build.sbt:32]: I was referencing the dependencies of hadoop-aws: https://mvnrepository.com/artifact/org.apache.hadoop/hadoop-aws/3.3.2 but yeah I try upgrading here

File: src/main/scala/io/github/spark_redshift_community/spark/redshift/RedshiftJDBCPushdownWrapper.scala (hunk 3 above)
Comment (dorisZ017) [reply src/main/scala/io/github/spark_redshift_community/spark/redshift/RedshiftJDBCPushdownWrapper.scala:90]: Will try adding some unit tests later

**Pull request merged**